docs/data/base/components/modal/modal.md

```diff
@@ -0,0 +1,128 @@
+<!-- If you are creating a modal dialog, the [`Dialog`](/material-ui/dialog/) component is better suited for this specific use case. -->
+
+`Modal` is a lower-level construct that is used in the following components:
+
+- [Dialog](/material-ui/react-dialog/)
+- [Drawer](/material-ui/react-drawer/)
+- [Menu](/material-ui/react-menu/)
+- [Popover](/material-ui/react-popover/)
+
+## Basic modal
+
+```js
+import ModalUnstyled from '@mui/base/ModalUnstyled';
+```
+
+{{"demo": "ModalUnstyled.js", "defaultCodeOpen": false }}
+
+## Nested modal
+
+> ⚠ **Note:** though it is possible to create nested modals—for example, a select modal within a dialog—stacking more than two at a time is discouraged.
+
+{{"demo": "NestedModal.js", "defaultCodeOpen": false}}
+
+## Transitions
+
+You can animate the open and close states of a modal with a transition component, as long as that component fulfills the following requirements:
+
+- Is a direct child descendent of the modal
+- Has an `in` prop—this corresponds to the open/close state
+- Calls the `onEnter` callback prop when the enter transition starts
+- Calls the `onExited` callback prop when the exit transition is completed
+  > The `onEnter` and `onExited` callbacks tell the modal to unmount the child content when closed and fully transitioned.
+
+`Modal` has built-in support for [react-transition-group](https://github.com/reactjs/react-transition-group):
+
+{{"demo": "TransitionsModal.js", "defaultCodeOpen": false}}
+
+You can also use [react-spring](https://github.com/pmndrs/react-spring) with the `Modal`, but it will require additional custom configuration:
+
+{{"demo": "SpringModal.js", "defaultCodeOpen": false}}
+
+## Performance
+
+The Modal component's content is unmounted when it is not open.
+This means that it will need to be re-mounted each time it is opened. If you are rendering expensive component trees inside your modal, and you want to optimize for interaction responsiveness, you can change this default behavior by enabling the keepMounted prop.
+
+You can also use the keepMounted prop if you want to make the content of the modal available to search engines (even when the modal is closed).
+
+```jsx
+<Modal keepMounted />
+```
+
+{{"demo": "KeepMountedModal.js", "defaultCodeOpen": false}}
+
+> You can use the `MuiModal-hidden` class to hide the modal when it is not open.
+
+As with any performance optimization, the `keepMounted` prop won't necessarily solve all of your problems.
+Explore other possible bottlenecks in performance where you could make more considerable improvements before implementing this prop.
+
+## Server-side modal
+
+React [doesn't support](https://github.com/facebook/react/issues/13097) the [`createPortal()`](https://reactjs.org/docs/portals.html) API on the server.
+
+In order to display a modal rendered on the server, you need to disable the portal feature with the `disablePortal` prop:
+
+{{"demo": "ServerModal.js", "defaultCodeOpen": false}}
```

Comment:
<img width="500" alt="Screen Shot 2022-03-18 at 11 41 57" src="https://user-images.githubusercontent.com/67129314/159024153-0817742b-c08f-4adc-b577-41d7814b47e9.png">

Small tweak might be needed here — text gets white in dark mode.

Reply:
Nice catch, will fix it.

Reply:
I've updated all demos for dark mode.